Galaxus.Functional.Tests/(Option)/OptionTests.cs

```diff
@@ -143,6 +144,110 @@ public void Option_MatchThrowsIfMatchArmIsNull()
         Assert.Throws<ArgumentNullException>(() => { _ = Option<int>.None.Match(v => v, null); });
     }
 
+    [Test]
+    public async Task Option_MatchAsyncSome_Works()
+    {
+        var some = "hello".ToOption();
+
+        {
+            var called = false;
+            await some.MatchAsync(async _ =>
```

Comment:
Instead of doing the assert inside the match, what about just setting a value and asserting that value after the match? I'd reckon its easier to follow the path, then.

(This applies to all the tests, of course)